trace/opencensus/trace/utils.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2017, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+UTF8 = 'utf-8'
+
+# Max length is 128 bytes for a truncatable string.
+MAX_LENGTH = 128
+
+
+def _get_truncatable_str(str_to_convert):
+    """Truncate a string if exceed limit and record the truncated bytes
+    count.
+    """
+    str_bytes = str_to_convert.encode(UTF8)
+    str_len = len(str_bytes)
```

Comment:
This seems kind of problematic, because you could convert a code point to a 2 or 3 byte utf-8 sequence, and then truncate the string in the middle of that sequence, creating an invalid string that causing an exception to be thrown.

But I think you can get around this by changing the code below to

```
truncated = str_bytes.decode(UTF8, errors='ignore')
```

This seemed to do the right thing in both Python 2 and 3.  Can you add a unit test where you call this on a string with 127 ascii characters followed by a some non-ascii characters?

I used the following code to play with truncation: `('\u2603\u2603'.encode('utf-8')[0:5]).decode('utf-8', errors='ignore')`.

Reply:
Thanks for the explanation! Added the unit test for that.